### PR TITLE
Use customer timeout in schema fetcher 

### PR DESCRIPTION
**Pull request by Cpaulyz**
## Description

Use customer timeout in schema fetcher

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/ClusterSchemaFetchExecutor.java [hunk 1]
```diff
@@ -46,42 +52,54 @@
...
+  private ExecutionResult executionStatement(
+      Long queryId, Statement statement, MPPQueryContext context) {
+    return coordinator.execute(
+        statement,
+        queryId,
+        context == null ? null : context.getSession(),
+        "",
+        ClusterPartitionFetcher.getInstance(),
+        schemaFetcher,
+        context == null || context.getQueryType().equals(QueryType.WRITE)
```
Comment (JackieTien97) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/ClusterSchemaFetchExecutor.java:85]: write should never  have a timeout.

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/AutoCreateSchemaExecutor.java [hunk 2]
```diff
@@ -54,21 +59,33 @@
...
     this.templateManager = templateManager;
-    this.statementExecutor = statementExecutor;
+    this.schemaFetcher = schemaFetcher;
+  }
+
+  private ExecutionResult executeStatement(Statement statement) {
+    return coordinator.execute(
+        statement,
+        SessionManager.getInstance().requestQueryId(),
+        null,
```
Comment (JackieTien97) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/AutoCreateSchemaExecutor.java:84]: try to gey `SessionInfo` here.

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/AutoCreateSchemaExecutor.java [hunk 3]
```diff
@@ -54,21 +59,33 @@
...
+        "",
+        ClusterPartitionFetcher.getInstance(),
+        schemaFetcher,
+        config.getQueryTimeoutThreshold());
```
Comment (JackieTien97) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/AutoCreateSchemaExecutor.java:88]: get timeout from context.

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/ClusterSchemaFetchExecutor.java (hunk 1 above)
Comment (JackieTien97) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/ClusterSchemaFetchExecutor.java:85]: Actually, we have already set the timeout field in `Context` for write to `Long.MAX_VALUE` in Coordinator.

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/ClusterSchemaFetchExecutor.java (hunk 1 above)
Comment (Cpaulyz) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/ClusterSchemaFetchExecutor.java:85]: > write should never have a timeout.

Yes, the statement must be query here but context may be WRITE type when inserting. So I use default timeout here.

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/AutoCreateSchemaExecutor.java (hunk 3 above)
Comment (Cpaulyz) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/AutoCreateSchemaExecutor.java:88]: This method will only be called when writing (for automatic schema creation).The type of context is WRITE, and its timeout is set to LONG.MAX_VALUE, which doesn't make any sense.

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/AutoCreateSchemaExecutor.java (hunk 2 above)
Comment (Cpaulyz) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/AutoCreateSchemaExecutor.java:84]: The statement here must be write type. SessionInfo seems to be useless in the write process.

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**